tools/rosout/rosout.cpp

```diff
@@ -82,10 +83,8 @@ class Rosout
 
   void init()
   {
-    bool disable_file_logging = false;
-    node_.getParamCached("/rosout/disable_file_logging", disable_file_logging);
-
-    if (!disable_file_logging)
+    const char* disable_file_logging = getenv("ROSOUT_DISABLE_FILE_LOGGING");
+    if (!disable_file_logging || !boost::iequals(disable_file_logging, "true"))
```

Comment:
This is new to me. Other than the allocation, is there any advantage to this over just instantiating a throwaway `std::string` and comparing that directly?

Reply:
`std::string` cannot be constructed with `nullptr`. So anyway, checking for `null` is required.

Reply:
gotcha, makes sense.

Reply:
The case insensitive comparison to `true` seems to be rather restrictive. I would suggest the opposite and if the string is empty or `0` (maybe "false", "off"?) consider the value to be `false`, otherwise `true`.